src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1469,7 +1469,8 @@ static ErrorCode getRefMismatchErrorCode(TypeKind type)
                 return true;
             }
 
-            if ((selectedMethod.HasUnsafeParameter() || selectedMethod.ReturnType.IsUnsafe()) && ReportUnsafeIfNotAllowed(syntax, diagnostics))
+            if ((selectedMethod.HasPointerOrFunctionPointerParameterType() || selectedMethod.ReturnType.IsPointerOrFunctionPointer())
```

Comment:
>selectedMethod.HasPointerOrFunctionPointerParameterType() || selectedMethod.ReturnType.IsPointerOrFunctionPointer() [](http://example.com/codeflow?start=17&length=115)

These conditions are no longer including arrays of pointers. Were those cases not used previously?

Reply:
Those were used. See the `UnsafeDelegateAssignment_*` tests. The scenarios involving arrays of pointers were previously reported as error, but no longer are.

Reply:
Are we testing constructed types such as `delegate List<int*[]> D();`? It looks like those cases should result in errors now.